parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -110,6 +131,11 @@ public void writeTo(OutputStream out) throws IOException {
     }
 
     @Override
+    public byte[] getBytesUnsafe() {
+      return getBytes();
```

Comment:
Yes. That option would definitely complicate things. But it would allow us to do fewer copies. Probably something we should do if we find out we're making too many copies in the future.